text/069-coq-roadmap.md

```diff
@@ -0,0 +1,270 @@
+the work, and any still unresolved questions to be answered.
+
+Rules :
+
+- An item cannot be part of the roadmap if it is not supported by at least
+  two persons, including at least one Coq maintainer to review the changes.
+- No one should be committed to work on more than two items at the same time.
+
+Each Coq Call will start with a roundtable about progress on items of the
+roadmap.
+
+Gaëtan Gilbert will be the technical coordinator of the roadmap, overseeing
+progress being made.
+
+Théo Zimmermann will be the editorial coordinator of the roadmap, proposing
+to add and remove items, to reflect the evolution of the project.
+
+## Priorities and resources
+
+### Kernel, theory
+
+#### Sort polymorphism
+
+- Gaëtan Gilbert, Nicolas Tabareau
+- 3 to 6 months
+- ongoing PR https://github.com/coq/coq/pull/17836 (will need a followup for inductives)
+
+Extension of universe polymorphism to sort qualities. It becomes possible to have global references quantified over 
+sort qualities eg `foo@{s | u | } : Type@{s | u}` which could be instantiated to `foo@{SProp | v} : SProp` 
+(the quantification syntax needs both `|` to avoid ambiguity).
+
+This should allow having for instance a common inductive for SProp, Prop and Type instantations of sigma types 
+(instead of the current `sigT` `sig` `ex` and variations depending on if each of the 2 parameters and the output are SProp).
+
+Eventually may allow using a common implementation for Type and Prop setoid rewriting machinery 
+(Morphisms vs CMorphisms, RelationClasses vs CRelationsClasses).
+
+May also be useful when doing Observational Type Theory.
+
+#### Algebraic universes
+
+- Matthieu Sozeau, Pierre-Marie Pédrot
+- 1 year
+
+#### Rewrite rules
+
+The goal is to add (unsafe) user-defined rewrite rules. This features allows
+users to add computation rules to axioms which can be useful for prototyping.
+It also allows for different kinds of computation rules with respect to what Coq
+currently permits: non-linearity, overlapping left-hand sides (*eg* one can write
+an addition on natural numbers that reduces on both sides: `0 + n` and `n + 0`
+both reducing to `n`).
+
+It would be deactivated by default and be optionally on by *eg* setting
+`Rewrite Rules On`. They should be supported by unification and the main
+reduction machines (not `vm_compute` and `native_compute` for now, Coq
+should give a warning when those are used with rewrite rules on). Rewrite
+rules would propagate to any module that requires the module that defines them.
+
+- Working Coq fork: https://github.com/Yann-Leray/coq (examples in
+https://github.com/Yann-Leray/coq/blob/rewrite-rules/test-suite/success/rewrule.v).
+- CEP: https://github.com/coq/ceps/pull/50
+- Yann Leray, Théo Winterhalter
+- 3 to 6 month
+
+#### Primitive projections
+
+Debate on the design to be had between Hugo Herbelin and Pierre-Marie Pédrot.
+
+### Surface language
+
+#### Ltac2
```

Comment:
Not sure what there is to say about this.

Reply:
We should I think at the very least try to clarify the objective of the current effort. E.g., make Ltac2 a realistic drop-in replacement for Ltac or whatever by providing fixes and minor improvements based on user feedback (it's better if it's your vision and Pierre-Marie's here rather than mine). What is your vision for the future of Ltac/Ltac2 for the next few years?

Reply:
I see "The goal is to get to the point where we can recommend new developments be Ltac2 only (no Ltac1)."  Is this new since the comment (and thus this thread can be resolved), or are you looking for something beyond that?

Reply:
This is new. I think it is both insufficiently detailed, and OK as a first step. But yes, let's mark this thread as resolved.